Java/core/src/main/java/com/amazon/randomcutforest/store/SmallLeafStore.java

```diff
@@ -37,20 +37,17 @@
  * The reference to the nodes will be offset by an amount that this ihe
  * capacity.
  */
-public class SmallLeafStore extends SmallIndexManager implements ILeafStore {
+public class SmallLeafStore extends IndexManager implements ILeafStore {
```

Comment:
Did we want to update these classes as well to has-an IndexManager from is-an Index manager?

Reply:
Great question -- we probably should. Given that we had to do that for PointStore and is-an became incredibly cumbersome from a constructor point of view, it is likely that down the road we will have to do the same for NodeStore.
LeafStores may not be needed at all even.